.github/workflows/main.yml

```diff
@@ -20,7 +20,7 @@ on:
       - submitted
 
 env:
-  XCLIM_TESTDATA_BRANCH: v0.41.0
+  XCLIM_TESTDATA_BRANCH: 2023.5.4
```

Comment:
Maybe rename `XCLIM_TESTDATA_BRANCH` to `XCLIM_TESTDATA_COMMIT_IDENTIFIER` so you can use branch, or tag, or commit?

Anyways, don't change if risky.  I think we understand what it can do.

Reply:
I'd rather settle on `BRANCH` as the use of this environment variable is to allow developers to test their branches using testing data in a different branch. This is currently documented as well.

Reply:
good for me.